build.ps1

```diff
@@ -330,29 +344,63 @@ function Invoke-VsDevShell($Arch) {
     Import-Module "$VSInstallRoot\Common7\Tools\Microsoft.VisualStudio.DevShell.dll"
     Enter-VsDevShell -VsInstallPath $VSInstallRoot -SkipAutomaticLocation -DevCmdArguments $DevCmdArguments
 
-    if ($WinSDKRoot) {
+    if (Test-Path script:WinSDKRoot) {
       # Using a non-installed Windows SDK. Setup environment variables manually.
-      $WinSDKVerIncludeRoot = "$WinSDKRoot\include\$WinSDKVersion"
+      $WinSDKVerIncludeRoot = "$WinSDKRoot\include\$WinSDKVersionRevisionZero"
       $WinSDKIncludePath = "$WinSDKVerIncludeRoot\ucrt;$WinSDKVerIncludeRoot\um;$WinSDKVerIncludeRoot\shared;$WinSDKVerIncludeRoot\winrt;$WinSDKVerIncludeRoot\cppwinrt"
-      $WinSDKVerLibRoot = "$WinSDKRoot\lib\$WinSDKVersion"
+      $WinSDKVerLibRoot = "$WinSDKRoot\lib\$WinSDKVersionRevisionZero"
 
-      $env:WindowsLibPath = "$WinSDKRoot\UnionMetadata\$WinSDKVersion;$WinSDKRoot\References\$WinSDKVersion"
+      $env:WindowsLibPath = "$WinSDKRoot\UnionMetadata\$WinSDKVersionRevisionZero;$WinSDKRoot\References\$WinSDKVersionRevisionZero"
       $env:WindowsSdkBinPath = "$WinSDKRoot\bin"
-      $env:WindowsSDKLibVersion = "$WinSDKVersion\"
-      $env:WindowsSdkVerBinPath = "$WinSDKRoot\bin\$WinSDKVersion"
-      $env:WindowsSDKVersion = "$WinSDKVersion\"
+      $env:WindowsSDKLibVersion = "$WinSDKVersionRevisionZero\"
+      $env:WindowsSdkVerBinPath = "$WinSDKRoot\bin\$WinSDKVersionRevisionZero"
+      $env:WindowsSDKVersion = "$WinSDKVersionRevisionZero\"
 
       $env:EXTERNAL_INCLUDE += ";$WinSDKIncludePath"
       $env:INCLUDE += ";$WinSDKIncludePath"
       $env:LIB += ";$WinSDKVerLibRoot\ucrt\$($Arch.ShortName);$WinSDKVerLibRoot\um\$($Arch.ShortName)"
       $env:LIBPATH += ";$env:WindowsLibPath"
       $env:PATH += ";$env:WindowsSdkVerBinPath\$($Arch.ShortName);$env:WindowsSdkBinPath\$($Arch.ShortName)"
-      $env:UCRTVersion = $WinSDKVersion
+      $env:UCRTVersion = $WinSDKVersionRevisionZero
       $env:UniversalCRTSdkDir = $WinSDKRoot
     }
   }
 }
 
+function Ensure-WindowsSDK {
+  # Assume we always have a default Windows SDK version available
+  if (-not $WinSDKVersion) { return }
+
+  # Check whether VsDevShell can already resolve the requested Windows SDK version
+  try {
+    Isolate-EnvVars { Invoke-VsDevShell $HostArch }
+    return
+  }
```

Comment:
Coddle the braces please

Reply:
I learned to do Egyptian braces but coddling will always be a challenge